builder/vmware/vmx/step_clone_vmx.go

```diff
@@ -18,49 +18,115 @@ type StepCloneVMX struct {
 	VMName    string
 }
 
+type vmxAdapter struct {
+	// The string portion of the address used in the vmx file
+	strAddr string
+	// Max address for adapter, controller, or controller channel
+	aAddrMax int
+	// Max address for device or channel supported by adapter
+	dAddrMax int
+}
+
+const (
+	// VMware Configuration Maximums - Virtual Hardware Versions 13/14
+	//
```

Comment:
Thanks for the explanation. That's fine to me. Given that this nested loop is iterating over tens of items rather than thousands, the speed sacrifice is likely to be negligible.